packages/flame/lib/src/geometry/ray2.dart

```diff
@@ -87,7 +101,9 @@ class Ray2 {
     final dot = _v2.dot(_v3);
     final t1 = _v2.cross(_v1) / dot;
     final t2 = _v1.dot(_v3) / dot;
-    if (t1 >= 0 && t2 >= 0 && t2 <= 1) {
+    // Using a small value above 0 just because of rounding errors later that
+    // might cause a ray to go in the wrong direction.
+    if (t1 > 0.00000000001 && t2 >= 0 && t2 <= 1) {
```

Comment:
`t1 = 0` corresponds to a ray whose origin lies on the segment. This is considered not intersecting now?

On the other hand, since `t1` is returned from the function, the upstream code can handle that case differently if it needs to.

Reply:
Yes, this fixed some bug. If I remember correctly it was when doing raytracing and doing any step later than the initial one, where origin is used from the intersection of the previous ray that was cast, then the ray will not necessarily continue in the correct direction, since it is counted as intersecting directly.

Reply:
Makes sense, thanks for the explanation.
Still, it would probably be better to have this logic upstream, in case someone else needs ray intersections for other purposes.

Reply:
Done!